samples/librats-server/server.c

```diff
@@ -0,0 +1,184 @@
+#include <librats/api.h>
+#include <librats/evidence.h>
+#include <librats/log.h>
+
+#define DEFAULT_PORT	1234
+#define DEFAULT_IP	"127.0.0.1"
+#define HEX_DUMP_SIZE	64
+#define BYTES		4
+#define MAX_HASH_LENGTH 1024
+
+//startup server
+int librats_server_startup(char *ip, int port, bool mutual)
+{
+	// create a socket
+	int sockfd = socket(AF_INET, SOCK_STREAM, 0);
+	if (sockfd < 0) {
+		RATS_ERR("Failed to call socket()");
+		return -1;
+	}
+
+	int reuse = 1;
+	if (setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR, (const void *)&reuse, sizeof(int)) < 0) {
+		RATS_ERR("Failed to call setsockopt()");
+		return -1;
+	}
+
+	struct sockaddr_in s_addr;
+	memset(&s_addr, 0, sizeof(s_addr));
+	s_addr.sin_family = AF_INET;
+	s_addr.sin_addr.s_addr = inet_addr(ip);
+	s_addr.sin_port = htons(port);
+
+	// bind socket
+	if (bind(sockfd, (struct sockaddr *)&s_addr, sizeof(s_addr)) == -1) {
+		RATS_ERR("Failed to call bind()");
+		return -1;
+	}
+
+	// listening, with up to 5 connections
+	if (listen(sockfd, 5) == -1) {
+		RATS_ERR("Failed to call listen()");
+		return -1;
+	}
+
+	while (1) {
+		RATS_INFO("Waiting for a connection ...\n");
+
+		// Receive connections from the client
+		struct sockaddr_in c_addr;
+		socklen_t size = sizeof(c_addr);
+
+		int connd = accept(sockfd, (struct sockaddr *)&c_addr, &size);
+		if (connd < 0) {
+			RATS_ERR("Failed to call accept()");
+			return -1;
+		}
+		RATS_INFO("get a client, ip:%s, port:%d\n", inet_ntoa(c_addr.sin_addr),
+			  ntohs(c_addr.sin_port));
+
+		// Receive the hash from the client and assign a value to your own hash
+		char buffer[MAX_HASH_LENGTH];
+		int len_recv = recv(connd, buffer, MAX_HASH_LENGTH, 0);
+		if (len_recv < 0) {
+			RATS_ERR("Failed to receive hash\n");
+		} else {
+			RATS_INFO("Successfully receive a hash of size %d\n", len_recv);
+		}
+		const char *hash = buffer;
+
+		// Collect evidence on the server side
+		attestation_evidence_t evidence;
+		rats_attester_err_t ret = librats_collect_evidence(&evidence, hash);
+		if (ret != RATS_ATTESTER_ERR_NONE) {
+			RATS_ERR("Librats collect evidence failed. Return code: %#x\n", ret);
+			return -1;
+		} else {
+			RATS_INFO("Successfully collect a evidence from server\n");
+		}
+
+		// Send evidence to client
+		int len_send = send(connd, (char *)&evidence, sizeof(attestation_evidence_t), 0);
+		if (len_send < 0) {
+			RATS_ERR("Failed to send evidence\n");
+			return -1;
+		} else {
+			RATS_INFO("Successfully send a evidence of size %d to client\n", len_send);
+		}
+
+		if (mutual) {
+			// Receive evidence from client
+			int len_evidence = sizeof(attestation_evidence_t);
+			char buffer[len_evidence];
+			attestation_evidence_t evidence_client;
+			int len_recv = recv(connd, buffer, sizeof(buffer), 0);
+			if (len_recv < 0) {
+				RATS_ERR("Failed to reveice evidence\n");
+				return -1;
+			}
+			if (len_recv == 0) {
+				RATS_ERR(
+					"The client side does not enable the mutual parameter. Please enable it! \n");
```

Comment:
Please format the log, some have the full stop and whitespace after the log, some not.